### PR TITLE
Updates to the cmake files.

### PR DESCRIPTION
**Pull request by dj2**
Move some common calls into cmake functions to remove duplication. Various other updates to the cmake scripts.

### CODE REVIEW

**Comment by dj2** [comment]
This refactoring was done to make it a bit simpler to turn on warnings in the various exe's and libraries. (see https://github.com/dj2/subspace/commit/a02e754eedc765e75716a5bfec6c4f02aacb5f12)

**Comment by danakj** [comment]
> This refactoring was done to make it a bit simpler to turn on warnings in the various exe's and libraries. (see [dj2@a02e754](https://github.com/dj2/subspace/commit/a02e754eedc765e75716a5bfec6c4f02aacb5f12))

Awesome! FWIW I had tried to enable warnings - but it feels like it only worked for MSVC. Welp now that I read it I guess I forgot the Wall 😂 https://github.com/chromium/subspace/blob/main/subspace/CMakeLists.txt#L209

File: subspace/CMakeLists.txt [hunk 1]
```diff
@@ -184,31 +187,14 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND
 endif()
 
 # Subspace test support
-set_target_properties(subspace_test_support PROPERTIES LINKER_LANGUAGE CXX)
-target_include_directories(subspace_test_support PRIVATE .)
-target_link_libraries(subspace_test_support
-    subspace
-)
+subspace_default_test_options(subspace_test_support)
```
Comment (danakj) [inline subspace/CMakeLists.txt:190]: nit: subspace_test_default_compile_options?

**Review by danakj — COMMENTED** [review]
2 requests

File: subspace/CMakeLists.txt (hunk 1 above)
Comment (dj2) [reply subspace/CMakeLists.txt]: done.

File: CMakeLists.txt [hunk 2]
```diff
@@ -20,18 +20,39 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
...
-set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+    set_target_properties(${TARGET} PROPERTIES LINKER_LANGUAGE CXX)
+    target_include_directories(${TARGET} PUBLIC ${PROJECT_SOURCE_DIR})
+endfunction()
+
+function(subspace_default_test_options TARGET)
+    subspace_default_compile_options(${TARGET})
+
+    target_include_directories(${TARGET} PRIVATE
+        ${PROJECT_SOURCE_DIR}/
```
[the remark this answers is not in the document]
Comment (dj2) [reply CMakeLists.txt]: Yup, good catch. Removed.

**Comment by danakj** [comment]
Thanks

**Pull request merged**